src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -600,14 +600,24 @@ def construct_payload(self):
 
     def validate_arguments(self):
         super().validate_arguments()
-        validate_create(self.get_argument_registry_identity(), self.get_argument_registry_pass(), self.get_argument_registry_user(), self.get_argument_registry_server(), self.get_argument_no_wait(), self.get_argument_source(), self.get_argument_repo(), self.get_argument_yaml(), self.get_argument_environment_type())
+        validate_create(self.get_argument_registry_identity(), self.get_argument_registry_pass(), self.get_argument_registry_user(), self.get_argument_registry_server(), self.get_argument_no_wait(), self.get_argument_source(), self.get_argument_artifact(), self.get_argument_repo(), self.get_argument_yaml(), self.get_argument_environment_type())
 
     def set_up_source(self):
-        if self.get_argument_source():
-            app, env = self._construct_app_and_env_for_source_or_repo()
+        from ._up_utils import (_validate_source_artifact_args)
+
+        source = self.get_argument_source()
+        artifact = self.get_argument_artifact()
+        _validate_source_artifact_args(source, artifact)
+        if artifact:
+            # Artifact is mostly a convenience argument provided to use --source specifically with a single artifact file.
+            # At this point we know for sure that source isn't set (else _validate_up_args would have failed), so we can build with this value.
+            source = artifact
```

Comment:
We Cannot use --source and --artifact together, right? There is a validate in `_validate_source_artifact_args`.
In line 614, we set `source=artifact`, then in line 620, 
app.run_source_to_cloud_flow(**source**, dockerfile, can_create_acr_if_needed=False, registry_server=self.get_argument_registry_server())

But in the `_post_process_for_repo`, we get `source ` from argument dict again.

I think you should update the source value in argument dict, too.

```
    def set_argument_source(self, source):
        self.set_param("source", source)
```

Reply:
Thanks, let me update this quickly